docs/source/contribution/cpu_features.rst

```diff
@@ -0,0 +1,218 @@
+    |-- abc_classification_train_method2_batch_fpt_cpu.cpp
+    |-- abc_classification_train_method1_impl.i
+    |-- abc_classification_train_method2_impl.i
+    |-- abc_classification_train_kernel.h
+    |-- abc_regression_predict_method1_batch_fpt_cpu.cpp
+    |-- abc_regression_predict_method1_batch_fpt_cpu.cpp
+    |-- abc_regression_predict_method1_impl.i
+    |-- abc_regression_predict_kernel.h
+    |-- abc_regression_train_method1_batch_fpt_cpu.cpp
+    |-- abc_regression_train_method2_batch_fpt_cpu.cpp
+    |-- abc_regression_train_method1_impl.i
+    |-- abc_regression_train_method2_impl.i
+    |-- abc_regression_train_kernel.h
+
+Alternative variant of the folder structure to avoid storing too many files within a single folder
+could be:
+
+::
+
+  cpp/daal/src/algorithms/abc/
+    |-- classification/
+    |     |-- abc_classification_predict_method1_batch_fpt_cpu.cpp
+    |     |-- abc_classification_predict_method1_impl.i
+    |     |-- abc_classification_predict_kernel.h
+    |     |-- abc_classification_train_method1_batch_fpt_cpu.cpp
+    |     |-- abc_classification_train_method2_batch_fpt_cpu.cpp
+    |     |-- abc_classification_train_method1_impl.i
+    |     |-- abc_classification_train_method2_impl.i
+    |     |-- abc_classification_train_kernel.h
+    |-- regression/
+          |-- abc_regression_predict_method1_batch_fpt_cpu.cpp
+          |-- abc_regression_predict_method1_impl.i
+          |-- abc_regression_predict_kernel.h
+          |-- abc_regression_train_method1_batch_fpt_cpu.cpp
+          |-- abc_regression_train_method2_batch_fpt_cpu.cpp
+          |-- abc_regression_train_method1_impl.i
+          |-- abc_regression_train_method2_impl.i
+          |-- abc_regression_train_kernel.h
+
+The names of the files stay the same in this case, just the folder layout differs.
+
+Further the purpose and contents of each file are to be described on the example of classification
+training task. For other types of the tasks the structure of the code is similar.
+
+\*_kernel.h
+-----------
+
+Those files contain the definitions of one or several template classes that define member functions that
+do the actual computations. Here is a variant of the ``Abc`` training algorithm kernel definition in the file
+`abc_classification_train_kernel.h`:
+
+.. include:: ../includes/cpu_features/abc-classification-train-kernel.rst
+
+Typical template parameters are:
+
+- ``algorithmFPType``  Data type to use in intermediate computations for the algorithm,
+                       ``float`` or ``double``.
+- ``method`` Computational methods of the algorithm. ``method1`` or ``method2`` in the case of ``Abc``.
+- ``cpu`` Version of the cpu-specific implementation of the algorithm, ``daal::CpuType``.
+
+Implementations for different methods are usually defined using partial class templates specialization.
+
+\*_impl.i
+---------
+
+Those files contain the implementations of the computational functions defined in `*_kernel.h` files.
```

Comment:
nit: Don't start the section with a pronoun. See the similar comment at the start of the `\*_kernel.h` section